### PR TITLE
Nicer Links in Docs

### PR DESCRIPTION
**Pull request by FG-TUM**
# Description

The Problem: Links in markdown files should work both in doxygen, the doc webpage, and on github.

Internal links don't work reliably on the webpage, hence all markdown files that are directly included in doxygen directly should use https links.

Repo-abolute paths in the documentation pages
- [x] README.md
- [x] CONTRIBUTING.md

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -106,82 +105,83 @@ Possible log levels are:`trace`, `debug`, `info`, `warn`, `err`, `critical`, `of
...
+* Adapt unit tests (e.g. expected number of iterations in [`AutoTunerTest::testAllConfigurations()`](/tests/testAutopas/tests/tuning/AutoTunerTest.cpp) and [`OptionTest::parseTraversalOptionsTest`](/tests/testAutopas/tests/options/OptionTest.h)).
 * Add new unit tests for your traversal.
-* Regenerate the `RuleLanguage.g4` via the `generateRuleLanguage.sh` script, both located in `src/autopas/tuning/tuningStrategy/ruleBasedTuning`
+* Regenerate the [`RuleLanguage.g4`](/src/autopas/tuning/tuningStrategy/ruleBasedTuning/RuleLanguage.g4) via the [`generateRuleLanguage.sh`](/src/autopas/tuning/tuningStrategy/ruleBasedTuning/generateRuleLanguage.sh) script, both located in [`ruleBasedTuning`](/src/autopas/tuning/tuningStrategy/ruleBasedTuning).
 
 ### Adding a new Container
-* Create a new container class under `src/autopas/containers/`.
-* Derive your new container from `src/autopas/containers/ParticleContainerInterface.h` or a more similar one.
-* Go to `src/autopas/options/ContainerOption.h`.
+* Create a new container class under [`src/autopas/containers/`](/src/autopas/containers/).
```
Comment (manishmishra6016) [inline .github/CONTRIBUTING.md:121]: As we are removing links from text everywhere, we could use something like `folder: containers`. I am not sure, just a suggestion. 

If seems feasible, also to be done at few other places.

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -106,82 +105,83 @@ Possible log levels are:`trace`, `debug`, `info`, `warn`, `err`, `critical`, `of
...
+* Adapt unit tests (e.g. expected number of iterations in [`AutoTunerTest::testAllConfigurations()`](/tests/testAutopas/tests/tuning/AutoTunerTest.cpp) and [`StringUtilsTest::parseContainerOptionsTest`](/tests/testAutopas/tests/utils/StringUtilsTest.cpp)).
 * Add new unit tests for your container.
-* Regenerate the `RuleLanguage.g4` via the `generateRuleLanguage.sh` script, both located in `src/autopas/tuning/tuningStrategy/ruleBasedTuning`
+* Regenerate the [`RuleLanguage.g4`](/src/autopas/tuning/tuningStrategy/ruleBasedTuning/RuleLanguage.g4) via the [`generateRuleLanguage.sh`](/src/autopas/tuning/tuningStrategy/ruleBasedTuning/generateRuleLanguage.sh) script, both located in [`ruleBasedTuning`](/src/autopas/tuning/tuningStrategy/ruleBasedTuning).
 
 ### Adding a new Tuning Strategy
-* Create a new tuning strategy class under `src/autopas/tuning/tuningStrategy`.
-* Derive your new strategy from `src/autopas/tuning/tuningStrategy/TuningStrategyInterface.h` or a more similar one.
-* Go to `src/autopas/options/TuningStrategyOption.h`.
+* Create a new tuning strategy class under [`src/autopas/tuning/tuningStrategy`](/src/autopas/tuning/tuningStrategy).
```
Comment (manishmishra6016) [inline .github/CONTRIBUTING.md:135]: same comment as above

File: README.md [hunk 3]
```diff
@@ -108,17 +107,17 @@ If you prefer `gdb`:
 As AutoPas is only a library, it is not able to run simulations by itself.
 We have, however, included a few example proxy applications in the **examples** directory.
 The examples include:
-* [md-flexible](examples/md-flexible): Molecular dynamics simulations with single centered Lennard-Jones particles.
+* [md-flexible](https://github.com/AutoPas/AutoPas/blob/master/examples/md-flexible): Molecular dynamics simulations with single centered Lennard-Jones particles.
 * Smoothed particle hydrodynamics simulations
```
Comment (manishmishra6016) [inline README.md:111]: Is there any link for this as well?

Comment (FG-TUM) [reply README.md]: yep -> 62ad2ea0e

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (FG-TUM) [reply .github/CONTRIBUTING.md:121]: can you post an example what exactly you mean? Not sure I understand...

Comment (manishmishra6016) [reply .github/CONTRIBUTING.md:121]: I meant something like 'Create a new container class under the folder: [containers](src/autopas/containers/)'. But it also looks fine as it is.

**Review by manishmishra6016 — APPROVED** [review]

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (manishmishra6016) [reply .github/CONTRIBUTING.md:121]: No, I think it is fine. In fact, it is more readable this way :)

**Pull request merged**